src/averageAges.js

```diff
@@ -38,6 +52,18 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const womenArr = withChildren
+    ? people.filter(
+      (person) =>
+        person.sex === 'f'
+          && people.some(child => child.mother === person.name)
+    )
+    : people.filter((person) => person.sex === 'f');
+  const womanAges = womenArr.map((person) => person.died - person.born);
+  const sumOfAges = womanAges.reduce((sum, age) => sum + age, 0);
+  const averageAge = sumOfAges / womanAges.length;
```

Comment:
create a separate helper function to calculate average ages

you do the same three times